src/monattack.cpp

```diff
@@ -4860,3 +4860,175 @@ bool mattack::dodge_check(monster *z, Creature *target){
+                _("<npcname> dodges the jet of black ink!") );
+        }
+        else if (u_see) {
+          add_msg(m_warning, 
+              _("The %s dodges the jet of black ink!"),
+              target->disp_name().c_str());
+        }
+
+    }
+
+}
+
+void mattack::tentacle_lash(monster *z, int index)
+{
+    Creature *target = z->attack_target();
+
+    int num_tentacles = 0;
+    for (const auto &ammo_entry : z->ammo) {
+        if (ammo_entry.first == "tentacle") {
+            num_tentacles += ammo_entry.second;
+        }
+    }
+    if (target == nullptr ||
+        !z->can_act() ||
+        num_tentacles == 0 ||
+        rl_dist(z->pos(), target->pos()) > 2 ||
+        !z->sees(*target)) {
+        return;
+    }
+
+    bool u_see = g->u.sees(*z);
+
+    const game_message_type msg_type = target->attitude_to(g->u) == Creature::A_FRIENDLY ? m_bad : m_warning;
+
+    int attacks_made = 0;
+
+    for (int i = 0; i < num_tentacles; i++) {
+
+        attacks_made++;
+
+        if (target->uncanny_dodge()) {
+            continue;
+        }
+        if (dodge_check(z, target)) {
+            if (u_see && !target->is_monster()) {
+                target->add_msg_player_or_npc(msg_type,
+                    _("The %s lashes tentacle at you, but you dodge it!"),
+                    _("The %s lashes tentacle at <npcname>!, but <npcname> dodges it!"),
+                    z->name().c_str());
+            }
+            else if (u_see) {
+                add_msg(msg_type, _
+                    ("The %1$s lashes tentacle at %2$s, but %2$s dodges it!"),
+                    z->name().c_str(), target->disp_name().c_str());
+            }
+            target->on_dodge(z, z->type->melee_skill * 2);
+            continue;
+        }
+        body_part hit = random_body_part();
+        int dam = rng(1, 10);
+        dam = target->deal_damage(z, hit, damage_instance(DT_BASH, dam)).total_damage();
+        if (dam > 0) {
+            if (u_see && !target->is_monster()) {
+                target->add_msg_player_or_npc(msg_type,
+                    _("The %1$s's tentacle lashes your %2$s!"),
+                    _("The %1$s's tentacle lashes <npcname>'s %2$s!"),
+                    z->name().c_str(),
+                    body_part_name_accusative(hit).c_str());
+            }
+            else if (u_see) {
+                add_msg(msg_type,
+                    _("The %1$s's tentacle lashes %2$s!"),
+                    z->name().c_str(),
+                    target->disp_name().c_str());
+            }
+            target->on_hit(z, hit, z->type->melee_skill);
+            target->check_dead_state();
+            if (target->is_dead_state()) {
+                break;
+            }
+        }
+        else {
+            if (u_see && !target->is_monster()) {
+                target->add_msg_player_or_npc(msg_type,
+                    _("The %1$s's tentacle lashes ineffectivelly at your %2$s"),
```

Comment:
"Ineffectively" - only one 'l'
Though this whole else isn't required - it's huge and not very informative. Would get really spammy with multi-tentacle attacks.
Instead of having a message per lash, there should be one at the end, summing up the assault.